cmd/snap/cmd_set_test.go

```diff
@@ -98,6 +98,34 @@ func (s *SnapSuite) TestSnapSetIntegrationJson(c *check.C) {
 	c.Assert(err, check.IsNil)
 }
 
+func (s *SnapSuite) TestSnapSetIntegrationUnsetWithExclamationMark(c *check.C) {
+	// mock installed snap
+	snaptest.MockSnap(c, string(validApplyYaml), &snap.SideInfo{
```

Comment:
We don't need this AFAICT - we only hit the mock server which will not look at any snap or state. Removing it (here and in the other tests) seems to be fine, the tests are still passing. The same for "validApplyYaml".